### DISCUSSION TITLE
use vupress to build  (downloadable) singlefile page

### DISCUSSION THREAD

**Discussion by s-light** [vuepress/core#1589] [Q&A]
Hello Community,

i would like to build a project where my source information are markdown documents.
these should be rendered to a single page (including each other in a tree like way..)

a big need is to have a single file that can be downloaded and viewed offline..

is this possible with the vuepress setup?

in another project i used quasar / vue.js with [vite-plugin-singlefile](https://github.com/richardtallent/vite-plugin-singlefile) to create a downloadable SPA.

any ideas and hints are welcome :-)

sunny greetings
stefan

**Comment by Mister-Hope** [comment]
I think the need is reasonable for.some cases, but a little werid. You want some reactive documents that should be opened with browser?

For offline usage (especially splitting into pages), you can just convert it to a pdf, simplest way should be printing through browser

For vuepress, we are not considering the feature you want in design. Inlining all tje resource is possible, but I am not sure you can implement it easily. The most difficulty is that it's hard to determine whether a script,is actualy imported nad used in page, as it's not ideal to inline all scripts

**Reply by s-light** [reply]
thanks for your reply Mister-Hope!

the idea is to have a little more than a pdf -  
a offline version that is a single-file-website that contains a custom search/filter for the content...

i will look into it a bit more - but yeah i know that my request is a very niche target for vuepress..

**Reply by meteorlxy** [reply]
The [local search plugin](https://ecosystem.vuejs.press/plugins/search/search.html) is possible for "custom search/filter for the content".

A "single-file-website" is a quite uncommon usage. Are you going to generate a manual for a product or something similar?

**Reply by s-light** [reply]
iam working in an educational project where we go to (German) schools for 3days and build /make/hack with electronics and microcontrollers with the pupils. for this we have a big material-case with us - and for every part in there is some (online) documentation available.
these documentation should be rebuild now.. (better examples and infos...)
we never know for sure that a school has good internet access - and sometimes its just down.. so one of my goals is to are prepared with an offline version on a usb-flash-drive - so we all can at least work with the basics without internet... 

i found the local-search plugin - yeah that should work.. but iam not sure if this is a good fit for my *single-file* concept idea..

and maybe i am just on the wrong way with my current concept idea..
in my idea i have all content on one big site - just to have a easier filtering and navigation..
(and if it is used online -  have the (long) wait time for page-load only once..)
